### ISSUE TITLE
feat: json maker

### ISSUE THREAD

**Issue by dag7dev** [LucianCrainic/ExamScraper#1]
Instead of producing a ton of files and folders with txt, it would be nice to have everything in a single json file, for each exam.

The file could be structured in a similar way to the JSQuizee ones:
```
  {
      "img": "if any.png",
      "question": "What's the favorite letter of the author?",
      "answers": [
          "The letter A",
          "The letter B",
          "The letter C",
          "The letter D"
      ],
      "correct": "d",
      "answers_have_code": 0,
      "code": ""
  },
```

Schema if needed:
```
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "img": {
      "type": "string"
    },
    "question": {
      "type": "string"
    },
    "answers": {
      "type": "array",
      "items": [
        {
          "type": "string"
        },
        {
          "type": "string"
        },
        {
          "type": "string"
        },
        {
          "type": "string"
        }
      ]
    },
    "correct": {
      "type": "string"
    },
    "answers_have_code": {
      "type": "integer"
    },
    "code": {
      "type": "string"
    }
  },
  "required": [
    "question",
    "answers",
    "correct",
  ]
}
```

TLDR;
- human and machine readable
- avoid a ton of files and folders, everything in a single file
- JSQuizee compatible (files are ready to be deployed to well-known [UniQuizzes](https://github.com/dag7dev/UniQuizzes))

**Comment by LucianCrainic** [comment]
I was thinking about it but the main problem is how to handle all the images. There is already a web scraper for software engineering course but it dosen't save images, that's why i wanted to make one that can store images too.

However my goal is to create a GUI (check the Jarvis repo) that users can use during the exam to search for questions that already exist, and i think the file system approach is more easy to handle.

I also want to implement the json version for the UniQuizzes but as i said before i have no idea how to handle all the images inside the json file. If you have a suggestion for it, i will gladly take it and implement.

**Comment by LucianCrainic** [comment]
```python
import json
import base64

data = {}
# open the image and encode it to base64
with open('image.png', mode='rb') as file:
    img = file.read()
# save the image to the data dictionary
data['img'] = base64.encodebytes(img).decode('utf-8')

# write the data to a json file
with open('data.json', mode='w') as file:
    json.dump(data, file)

# read the data from the json file
with open('data.json', mode='r') as file:
    data = json.load(file)
    # decode the image
    img = base64.decodebytes(data['img'].encode('utf-8'))
    # write the image
    with open('new_image.png', mode='wb') as file:
        file.write(img)
```
I have this possible solution, to encode all the images to base64 and then you can decode them on the UniQuizzes website. Let me know if you have a different idea for this.

**Comment by dag7dev** [comment]
I hadn't thought about base64, even in JSQuizee, that's a great idea! In this way we don't have to store them as separate images, and we don't care about providing a filename.

**Comment by LucianCrainic** [comment]
sounds great. i'll implement it asap. Thank you for the suggestion !
